src/soilbiogeochem/SoilBiogeochemCompetitionMod.F90

```diff
@@ -722,6 +759,78 @@ subroutine SoilBiogeochemCompetition (bounds, num_soilc, filter_soilc,num_soilp,
+               if (h2osoi_vol(c,j)  < 0.30_r8 .and. pfactor_vr(c,j) .eq. 1._r8) then
+
+   ! If change in soil moisture is > 0.005 (0.5% v/v) and there is precipitation, then start pulsing
+                  if ( h2osoi_diff(c,j) > 0.005_r8 ) then
+ ! Initialize new pulse factor (dry period hours)
+                     if ( ldry_vr(c,j) > 72._r8 ) then
+                        pfactor_vr(c,j) = 13._r8 *LOG(ldry_vr(c,j)) - 53.6_r8
+                     else
+                        pfactor_vr(c,j)=1._r8
+                     endif
+! If dry period < ~3 days then no pulse.  "13._r8 *LOG(ldry(c,j))" becomes > 53.6 at > 72 hours
+                       ! Reinitialize dry period
+                     ldry_vr(c,j) = 0._r8
+                  else ! If no rain (i.e.,  change in soil moisture is < 0.5%)
+                     ! Add one timestep (in hours) to dry period
+                     ldry_vr(c,j) = ldry_vr(c,j) + dthr
+                  endif
+                 else
+                    pfactor_vr(c,j) = pfactor_vr(c,j) * exp( -0.068_r8 * dthr )
+
+              ! Update dry period
+                    If ( h2osoi_vol(c,j) < 0.30_r8 ) then
+                       ldry_vr(c,j)=ldry_vr(c,j) + dthr
+                    endif
+                 endif
+
+                 ! If end of pulse
+                 if ( pfactor_vr(c,j) < 1._r8 ) then
+                    pfactor_vr(c,j) = 1._r8
+                 endif
+                 ! Apply the pulse factor to NOx nitrification emission
+                 pot_f_nox_nit_vr(c,j)   = pot_f_nox_nit_vr(c,j) * pfactor_vr(c,j)
+
+                 ! final NOx cannot be larger than available nitrogen from nitrification, considering the N2O nitrified
+                 f_nox_nit_vr(c,j)   = min(pot_f_nox_nit_vr(c,j), max(0._r8,f_nit_vr(c,j)-f_n2o_nit_vr(c,j)))
+
+                 !soil NOx denitrification (DayCent code)
+                 pot_f_nox_denit_vr(c,j) = f_n2o_denit_vr(c,j) * nox_n2o_ratio_vr(c,j)  
+                 
+                 ! final NOx from denitrification cannot be larger than available nitrogen from denitrification
+                 f_nox_denit_vr(c,j)   = min(pot_f_nox_denit_vr(c,j), max(0._r8, f_n2o_denit_vr(c,j)))
+                
+                 
+                 !some of the NOx denit needs to come from f_n2 as f_NOx+f_N2+f_N2O cannot be larger than Fdenit)
+                 f_n2_denit_vr(c,j)=f_n2_denit_vr(c,j)-f_nox_denit_vr(c,j)
+                 
+              
+               !remove NOx trapped in canopy, ie, NO that makes it into the atmosphere. N in canopy is assumed to returned to the soil for now.            
+               !NOx flux above canopy
+               !sanity check, make sure CRF ranges from 0 to 1
+                 crf_drydep_col(c)= min(1._r8, max(0._r8, crf_drydep_col(c)))
+
+                 f_nox_nit_atmos_vr(c,j) =f_nox_nit_vr(c,j)* crf_drydep_col(c)
+                 f_nox_denit_atmos_vr(c,j)=f_nox_denit_vr(c,j)*crf_drydep_col(c)
```

Comment:
I'm confused where the NOx that's trapped in the canopy come back to the soil.  Does this just occur because only the `f_nox_nit_atmos_vr_col` is actually removed from the `smin_no3_vr_col` pool in N_StateUpdate (below)?

```
ns%smin_no3_vr_col(c,j) = ns%smin_no3_vr_col(c,j) - nf%f_nox_nit_atmos_vr_col(c,j) * dt
```

Reply:
Correct, only the NO that is not trapped in the canopy exits the system and is paased into the atmosphere. Both f_nox_nit_atmos and f_nox_denit_atmos leave the system. NO trapped in the canopy ideally should contribute to plant nitrogen uptake, but  there are uncertainties in how plants process that nitrogen and how we can parameterize these processes.